spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -240,15 +253,171 @@ else if (messageListener != null) {
 						.maxNumMessages(containerProperties.getMaxNumMessages())
 						.maxNumBytes(containerProperties.getMaxNumBytes())
 						.timeout(containerProperties.getBatchTimeoutMillis(), TimeUnit.MILLISECONDS).build();
+
+				Set<String> topicNames = (Set<String>) propertiesToConsumer.remove("topicNames");
+				Map<String, String> properties = (Map<String, String>) propertiesToConsumer.remove("properties");
+				Map<String, String> subscriptionProperties = (Map<String, String>) propertiesToConsumer
+						.remove("subscriptionProperties");
+
+				ConsumerBuilderCustomizer<T> customizer = builder -> {
+					loadConf(builder, propertiesToConsumer);
+					builder.batchReceivePolicy(batchReceivePolicy);
+				};
 				this.consumer = getPulsarConsumerFactory().createConsumer((Schema) containerProperties.getSchema(),
-						batchReceivePolicy, propertiesToConsumer);
+						topicNames, properties, subscriptionProperties, Collections.singletonList(customizer));
 				Assert.state(this.consumer != null, "Unable to create a consumer");
 			}
 			catch (PulsarClientException e) {
 				DefaultPulsarMessageListenerContainer.this.logger.error(e, () -> "Pulsar client exceptions.");
 			}
 		}
 
+		/*
+		 * Method to work around the issue that ConsumerBuilder::loadConf sets JsonIgnored
+		 * fields of ConsumerConfigurationData to null. To be removed once
+		 * https://github.com/apache/pulsar/issues/11646 is fixed.
+		 */
+		@SuppressWarnings("unchecked")
+		private static <T> void loadConf(ConsumerBuilder<T> builder, Map<String, Object> properties) {
```

Comment:
this would need to be fully tested

Reply:
Bummer - this is unfortunate. Could we not instead remove+remember the problematic properties, then call the standard loadConf, and finally apply the problem fields manually after the fact. It seems like it would reduce this additional area of mapping.

Reply:
The problem is that if you have for instance a `deadLetterPolicy` in the current builder, calling `loadConf` will erase it. And we don't have access to the inner values of the builder to remember them before.

Reply:
Yep, I remember the original issue. The difference here is ...

>  And we don't have access to the inner values of the builder to remember them before.

Is the only problem field the DLP? If so I would be in favor of digging that value out of the builder via reflection than have to do the mapping w/ all of the fields.

Reply:
No DLP is not the only one. I can try to do the save/load/replace trick for all of them

Reply:
Actually we can't call `loadConf` because there will be a crash if DLP was set before on the builder.
So we don't have a choice but to map manually. We can remove this when `loadConf` is fixed.
IIUC `BeanUtils.copyProperties` copies fields with the same name from one struct to the other which is not what is done here.